### PR TITLE
feat: add delete address functionality

### PR DESCRIPTION
**Pull request by gkartalis**
The type of this PR is: **Feature**

This PR resolves [PURCHASE-2787]

### Description

- Allow an address to be deleted when a user selects the delete text link on the card by adding a deleteAddress mutation

https://user-images.githubusercontent.com/21178754/125631970-70c8fe18-44f1-4223-9a1f-91f913da82e7.mp4

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a CHANGELOG.yml entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Enables users to delete saved addresses

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[PURCHASE-2787]: https://artsyproduct.atlassian.net/browse/PURCHASE-2787

### CODE REVIEW

**Comment by gkartalis** [comment]
Noticed in the video that when removing address the unmounting of the element is not the smoothest. Maybe we can address that in another ticket after syncing with design?

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Enables users to delete saved addresses)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a18a599d8de398a2394c95e5e08c19fb8e6a2ae2
</p>

File: src/lib/Scenes/SavedAddresses/SavedAddresses.tsx [hunk 1]
```diff
@@ -156,6 +167,54 @@ export const SavedAddressesContainer = createRefetchContainer(
...
+      const errors = data?.deleteUserAddress?.userAddressOrErrors.errors
+      if (errors) {
+        onError(errors.map((error) => error.message).join(", "))
+      } else {
+        onSuccess()
+      }
+    },
+  })
+}
+
```
Comment (jo-rs) [inline src/lib/Scenes/SavedAddresses/SavedAddresses.tsx:217]: I'd say, let's move this guy to it's own file so we can then reuse it on the edit form?

File: src/lib/Scenes/SavedAddresses/SavedAddresses.tsx [hunk 2]
```diff
@@ -39,7 +41,16 @@ const SavedAddresses: React.FC<{ me: SavedAddresses_me; relay: RelayRefetchProp
 
   const onPressEditAddress = () => null
 
-  const onPressDeleteAddress = () => null
+  const onPressDeleteAddress = (addressId: string) => {
+    deleteUserAddress(
+      addressId,
+      () =>
+        relay.refetch({
+          first: 3,
```
Comment (jo-rs) [inline src/lib/Scenes/SavedAddresses/SavedAddresses.tsx:49]: Is this `3` trivial? 🤔

**Review by jo-rs — CHANGES_REQUESTED** [review]
It looks great 💯 
Let's move the mutation to its own file, so we can reuse it. And, are there any tests that we can add?

**Comment by jo-rs** [comment]
> Noticed in the video that when removing address the unmounting of the element is not the smoothest. Maybe we can address that in another ticket after syncing with design?

I think once we add the toast in, it will feel a bit nicer 🤞

**Review by jo-rs — APPROVED** [review]
Nice! 🚀

**Pull request merged**